lib/rubocop/cop/rspec/factory_bot/factory_class_name.rb

```diff
@@ -6,6 +6,11 @@ module RSpec
       module FactoryBot
         # Use string value when setting the class attribute explicitly.
         #
+        # Although FactoryBot allows passing a class, passed application models
+        # are loaded before the application itself gets loaded.
```

Comment:
The original motivation was to avoid loading those classes if they are not needed, and let autoload to do its job. I'm not 100% convinced this works as expected though.

Reply:
Do you mean that adding a description like "When not to use it" or "Justification" is good?

Reply:
There's no common practice, some cops only explain what they do, [some](https://github.com/rubocop-hq/rubocop-rspec/blob/master/manual/cops_rspec.md#rspecexamplelength) why they do it. However, if there's a motivation, it needs to be solid.

Reply:
First of all, I am OK with being more specific regarding both the what and the why. Secondly, this is low profile cop that may not affect you ever.

The main why is speed, lazy-loading of course. 
However, my motivation behind this cop comes from personal experience on (let's say) bad code, legacy or not, in combination with external libraries, gems etc. There are cases where gems/engines evaluate application files before the applications get loaded. This is something potentially wrong, bad etc.
One could make use of this cop if needed to avoid pre-loading application files **at least** from factories.

Reply:
@jfragoulis Thanks for your detail feedback! Then, I try to change the description as follows. I'm not native in English, so its wording may feel strange. I would be grateful if you could point out some strange parts.

Here is my draft:

> This cop would promote faster tests by lazy-loading of application files. Also, this could help you suppress potential bugs in combination with external libraries by avoiding a preload of application files from the factory files.

Reply:
This is perfect.

Reply:
Thank you. I've pushed the change via d4e1e5e56d7ad40538d43a29a5576a2963e8ab11.

Could you please re-review?